### ISSUE TITLE
Insta ban on failed to connect

### ISSUE THREAD

**Issue by nholland94** [CodaProtocol/coda#2349]
Currently, the codebase will instantly ban peers it cannot connect to. However, this is too harsh as the protocol allows nodes to go offline. The correct behavior should be a small decrease in trust mixed with removing that peer from our list of peers.

**Comment by enolan** [comment]
What we want here is the ability to "kick" people in addition to banning them. I suppose we could hack that in to our DHT by just banning the peer and immediately unbanning it. @cmr is this something that makes sense in libp2p?

**Comment by nholland94** [comment]
Shouldn't "kicking" a peer just be removing it from the internal peer list we acquire from kademlia?

**Comment by cmr** [comment]
@enolan I don't follow what you mean by kick - disconnect them without forbidding reconnections?

**Comment by psteckler** [comment]
What is the symptom of "cannot connect to"? ECONNREFUSED?

**Comment by enolan** [comment]
@cmr remove them from our routing table without banning them.

@psteckler we get:
```
((rpc_error (Connection_closed (\"EOF or connection closed\")))
  (connection_description <created-directly>) (rpc_tag __Versioned_rpc.Menu)
  (rpc_version 1))```

Working on this now.

**Comment by psteckler** [comment]
Addressed by #2365.
